redbeams/src/main/java/com/sequenceiq/redbeams/converter/database/DatabaseV4RequestToDatabaseConfigConverter.java

```diff
@@ -0,0 +1,60 @@
+package com.sequenceiq.redbeams.converter.database;
+
+
+import static com.sequenceiq.cloudbreak.common.type.APIResourceType.RDS_CONFIG;
+
+import java.util.Date;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.google.common.base.Strings;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.DatabaseVendor;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.ResourceStatus;
+import com.sequenceiq.cloudbreak.common.converter.MissingResourceNameGenerator;
+import com.sequenceiq.cloudbreak.converter.AbstractConversionServiceAwareConverter;
+import com.sequenceiq.redbeams.api.endpoint.v4.database.request.DatabaseV4Request;
+import com.sequenceiq.redbeams.domain.DatabaseConfig;
+import com.sequenceiq.redbeams.util.DatabaseVendorUtil;
+
+@Component
+public class DatabaseV4RequestToDatabaseConfigConverter  extends AbstractConversionServiceAwareConverter<DatabaseV4Request, DatabaseConfig> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseV4RequestToDatabaseConfigConverter.class);
+
+    @Inject
+    private MissingResourceNameGenerator missingResourceNameGenerator;
+
+    @Override
+    public DatabaseConfig convert(DatabaseV4Request source) {
+        DatabaseConfig databaseConfig = new DatabaseConfig();
+        if (Strings.isNullOrEmpty(source.getName())) {
+            databaseConfig.setName(missingResourceNameGenerator.generateName(RDS_CONFIG));
+        } else {
+            databaseConfig.setName(source.getName());
+        }
+        databaseConfig.setDescription(source.getDescription());
+        databaseConfig.setConnectionURL(source.getConnectionURL());
+
+        DatabaseVendor databaseVendor = DatabaseVendorUtil.getVendorByJdbcUrl(source).get();
+        databaseConfig.setDatabaseEngine(databaseVendor);
+        databaseConfig.setConnectionDriver(databaseVendor.connectionDriver());
+        databaseConfig.setConnectionUserName(source.getConnectionUserName());
+        databaseConfig.setConnectionPassword(source.getConnectionPassword());
+        databaseConfig.setCreationDate(new Date().getTime());
```

Comment:
Nit: just use System.currentTimeMillis()

Reply:
Thank you. Yes, copy/paste ...
We have class Clock in common that has getCurrentTimeMillis functionality.